src/L1/rollup/ScrollChain.sol

```diff
@@ -696,115 +721,25 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
-    }
-
-    /// @notice Add an account to the sequencer list.
-    /// @param _account The address of account to add.
-    function addSequencer(address _account) external onlyOwner {
-        // @note Currently many external services rely on EOA sequencer to decode metadata directly from tx.calldata.
-        // So we explicitly make sure the account is EOA.
-        if (_account.code.length > 0) revert ErrorAccountIsNotEOA();
-
-        isSequencer[_account] = true;
-
-        emit UpdateSequencer(_account, true);
-    }
-
-    /// @notice Remove an account from the sequencer list.
-    /// @param _account The address of account to remove.
-    function removeSequencer(address _account) external onlyOwner {
-        isSequencer[_account] = false;
-
-        emit UpdateSequencer(_account, false);
     }
 
-    /// @notice Add an account to the prover list.
-    /// @param _account The address of account to add.
-    function addProver(address _account) external onlyOwner {
-        // @note Currently many external services rely on EOA prover to decode metadata directly from tx.calldata.
-        // So we explicitly make sure the account is EOA.
-        if (_account.code.length > 0) revert ErrorAccountIsNotEOA();
-        isProver[_account] = true;
-
-        emit UpdateProver(_account, true);
-    }
-
-    /// @notice Add an account from the prover list.
-    /// @param _account The address of account to remove.
-    function removeProver(address _account) external onlyOwner {
-        isProver[_account] = false;
-
-        emit UpdateProver(_account, false);
-    }
-
-    /// @notice Update the value of `maxNumTxInChunk`.
-    /// @param _maxNumTxInChunk The new value of `maxNumTxInChunk`.
-    function updateMaxNumTxInChunk(uint256 _maxNumTxInChunk) external onlyOwner {
-        uint256 _oldMaxNumTxInChunk = maxNumTxInChunk;
-        maxNumTxInChunk = _maxNumTxInChunk;
-
-        emit UpdateMaxNumTxInChunk(_oldMaxNumTxInChunk, _maxNumTxInChunk);
-    }
-
-    /// @notice Pause the contract
-    /// @param _status The pause status to update.
-    function setPause(bool _status) external onlyOwner {
-        if (_status) {
-            _pause();
-        } else {
-            _unpause();
-        }
-    }
-
-    /**********************
-     * Internal Functions *
-     **********************/
-
     /// @dev Internal function to do common checks before actual batch committing.
     /// @param _parentBatchHeader The parent batch header in calldata.
     /// @param _chunks The list of chunks in memory.
+    /// @param lastCommittedBatchIndex The index of last committed batch.
     /// @return _parentBatchHash The batch hash of parent batch header.
     /// @return _batchIndex The index of current batch.
     /// @return _totalL1MessagesPoppedOverall The total number of L1 messages popped before current batch.
-    function _beforeCommitBatch(bytes calldata _parentBatchHeader, bytes[] memory _chunks)
+    function _beforeCommitBatch(
```

Comment:
There's only 1 call site for `_beforeCommitBatch` in line 567 and we always call it with `lastCommittedBatchIndex=0`. Why is that?
```
(bytes32 _parentBatchHash, uint256 _batchIndex, uint256 _totalL1MessagesPoppedOverall) = _beforeCommitBatch(
            _parentBatchHeader,
            _chunks,
            0
);
```

Reply:
fixed, one of the left todo.